sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/encoders/RowEncoderSuite.scala

```diff
@@ -185,6 +185,20 @@ class RowEncoderSuite extends SparkFunSuite {
     assert(encoder.serializer.head.nullable == false)
   }
 
+  test("RowEncoder should support array as the external type for ArrayType") {
```

Comment:
We have `encodeDecodeTest` in `RowEncoderSuite`, but this is the special case. ideally all types that allow more than one external types need individual test, i.e. decimal type, struct type, array type.